front-end/src/screens/Settings/setCredentials.tsx

```diff
@@ -0,0 +1,260 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { web3Accounts, web3Enable, web3FromSource } from '@polkadot/extension-dapp';
+import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
+import { stringToHex } from '@polkadot/util';
+import React, { useContext, useEffect, useState } from 'react';
+import { FieldError,useForm } from 'react-hook-form';
+import { DropdownProps } from 'semantic-ui-react';
+
+import ExtensionNotDetected from '../../components/ExtensionNotDetected';
+import { UserDetailsContext } from '../../context/UserDetailsContext';
+import { useSetCredentialsConfirmMutation, useSetCredentialsStartMutation } from '../../generated/graphql';
+import { handleTokenChange } from '../../services/auth.service';
+import AccountSelectionForm from '../../ui-components/AccountSelectionForm';
+import Button from '../../ui-components/Button';
+import FilteredError from '../../ui-components/FilteredError';
+import { Form } from '../../ui-components/Form';
+import Loader from '../../ui-components/Loader';
+import Modal from '../../ui-components/Modal';
+import getEncodedAddress from '../../util/getEncodedAddress';
+import messages from '../../util/messages';
+
+const APPNAME = process.env.REACT_APP_APPNAME || 'polkassembly';
+
+const SetCredentials = (): JSX.Element => {
+	const [error, setErr] = useState<Error | null>(null);
+	const [showModal, setShowModal] = useState(false);
+	const [address, setAddress] = useState<string>('');
+	const [accounts, setAccounts] = useState<InjectedAccountWithMeta[]>([]);
+	const [isAccountLoading, setIsAccountLoading] = useState(true);
+	const [extensionNotFound, setExtensionNotFound] = useState(false);
+	const [accountsNotFound, setAccountsNotFound] = useState(false);
+	const { errors, handleSubmit, register } = useForm();
+	const [setCredentialsStartMutation] = useSetCredentialsStartMutation();
+	const [setCredentialsConfirmMutation, { loading }] = useSetCredentialsConfirmMutation();
+	const currentUser = useContext(UserDetailsContext);
+
+	useEffect(() => {
+		if (!accounts.length) {
+			getAccounts();
+		}
+	}, [accounts.length]);
+
+	const getAccounts = async (): Promise<void> => {
+		const extensions = await web3Enable(APPNAME);
+
+		if (extensions.length === 0) {
+			setExtensionNotFound(true);
+			setIsAccountLoading(false);
+			return;
+		} else {
+			setExtensionNotFound(false);
+		}
+
+		const accounts = await web3Accounts();
+
+		if (accounts.length === 0) {
+			setAccountsNotFound(true);
+			setIsAccountLoading(false);
+			return;
+		} else {
+			setAccountsNotFound(false);
+		}
+
+		accounts.forEach((account) => {
```

Comment:
I don't think we need to get the all the web3 accounts here. All we need is any account that is linked with the user. Narrowing things down to the default account sounds good enough

Reply:
We need to check if default Address is available in the extension and get the signer.